src/main/kotlin/net/developerden/devdenbot/listener/DiscordTokenMessageListener.kt

```diff
@@ -0,0 +1,53 @@
+package net.developerden.devdenbot.listener
+
+import net.developerden.devdenbot.discord.BOT_COMMANDS_CHANNEL_ID
+import net.developerden.devdenbot.extensions.await
+import net.developerden.devdenbot.serialization.DDBConfig
+import net.developerden.devdenbot.util.handleEachIn
+import net.developerden.devdenbot.util.listenFlow
+import net.developerden.devdenbot.util.scope
+import net.dv8tion.jda.api.JDA
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import javax.inject.Inject
+
+class DiscordTokenMessageListener @Inject constructor(
+    private val ddbConfig: DDBConfig,
+    ) : EventListener {
+
+    companion object {
+        private val log by log()
+
+        internal val tokenRegex = Regex("[MN][A-Za-z\\d]{23}\\.[\\w-]{6}\\.[\\w-]{27}\n")
+    }
+
+    private suspend fun onGuildMessageReceived(event: GuildMessageReceivedEvent) {
+        if (event.author.isBot) {
+            return
+        }
+        if (event.channel.idLong == BOT_COMMANDS_CHANNEL_ID) {
+            return
+        }
+        val rawText = event.message.contentRaw
+        if (rawText.startsWith("${ddbConfig.prefix}keep")) {
+            return
+        }
+        if (rawText.startsWith("/run")){
+            return
+        }
+        if (!rawText.contains(tokenRegex)) {
+            return
+        }
+        if (rawText.contains(tokenRegex)) {
```

Comment:
this will always be true because you've already checked `!rawText.contains(tokenRegex)`